### PR TITLE
[chore] clean up validation of SAPM exporter

### PR DESCRIPTION
**Pull request by atoulme**
Clean up of the `validate` and `Validate` functions used by sapmexporter.
This PR merges them, and moves the editing of the endpoint to use https scheme if not explicitly used closer to using the endpoint.

### CODE REVIEW

**Comment by dmitryax** [comment]
PTAL at the failing tests

File: exporter/sapmexporter/config.go [hunk 1]
```diff
@@ -61,33 +60,29 @@ type Config struct {
...
-		e.Scheme = defaultEndpointScheme
-	}
-	c.Endpoint = e.String()
-	return nil
-}
-
-func (c *Config) Validate() error {
 	if err := c.QueueSettings.Validate(); err != nil {
 		return fmt.Errorf("sending_queue settings has invalid configuration: %w", err)
 	}
```
Comment (dmitryax) [inline exporter/sapmexporter/config.go:73]: AFAIR this is also not needed anymore, but please verify before removing

File: exporter/sapmexporter/exporter_test.go [hunk 2]
```diff
@@ -53,14 +53,6 @@ func TestCreateTracesExporter(t *testing.T) {
 	assert.NoError(t, te.Shutdown(context.Background()), "trace exporter shutdown failed")
 }
 
-func TestCreateTracesExporterWithInvalidConfig(t *testing.T) {
-	cfg := &Config{}
-	params := exportertest.NewNopCreateSettings()
-	te, err := newSAPMTracesExporter(cfg, params)
```
Comment (dmitryax) [inline exporter/sapmexporter/exporter_test.go:59]: Is the `Validate` with no endpoint covered somewhere else?

**Comment by dmitryax** [comment]
LGTM , just couple nits

File: exporter/sapmexporter/config.go (hunk 1 above)
Comment (atoulme) [reply exporter/sapmexporter/config.go:73]: if that's true I can delete even more code ;)

File: exporter/sapmexporter/exporter_test.go (hunk 2 above)
Comment (atoulme) [reply exporter/sapmexporter/exporter_test.go:59]: yes, tested here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/eb8a8309909b4a909a307666eefab6657b282428/exporter/sapmexporter/config_test.go#L100

File: exporter/sapmexporter/config.go (hunk 1 above)
Comment (atoulme) [reply exporter/sapmexporter/config.go]: Confirmed. I have a test showing the component.ValidateConfig method will test queue settings.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
LGTM. Thank you!

**Pull request merged**